### PR TITLE
Add phonemized LibriTTS ASR recipe

### PR DESCRIPTION
**Pull request by akreal**
## What?

ASR recipe for LibriTTS with phonemized transcriptions.

## Why?

As per discussion in https://github.com/espnet/espnet/issues/5393 

## See also

The system is similar to the system described in [this](https://www.voiceprivacychallenge.org/results-2022/docs/3___T04.pdf) paper.

I'll add results and model link in a week or two.

### CODE REVIEW

**Comment by sw005320** [comment]
Many thanks!
This is really useful.

File: egs2/libritts/asr1/run.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+valid_set="dev"
+test_sets="test-clean test-other dev-clean dev-other"
+
+asr_config=conf/train_asr.yaml
+inference_config=conf/decode_asr.yaml
+
+./asr.sh \
+    --lang en \
+    --ngpu 2 \
+    --nbpe 100 \
```
Comment (ftshijt) [inline egs2/libritts/asr1/run.sh:18]: Is 100 bpe size an empirical good number?

File: egs2/libritts/asr1/local/phonemize_dir.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+with open(f"{idir}/text", encoding="utf-8") as itext, open(
+    f"{idir}/text.phn", "w", encoding="utf-8"
+) as otext:
+    for line in itext:
+        utt, text = line.strip("\n").split(" ", maxsplit=1)
+        tokens = tokenizer.text2tokens(text)
+        text_phn = "".join(tokens).replace("<space>", " ")
+        otext.write(f"{utt} {text_phn}\n")
+
+os.replace(f"{idir}/text.phn", f"{idir}/text")
```
Comment (ftshijt) [inline egs2/libritts/asr1/local/phonemize_dir.py:21]: maybe consider keep the original text

**Review by ftshijt — COMMENTED** [review]
Thanks for the contribution. Some minor comments/questions:

File: egs2/libritts/asr1/run.sh (hunk 1 above)
Comment (akreal) [reply egs2/libritts/asr1/run.sh:18]: Yes, it provided the best average PER in preliminary experiments:

<table>
  <tr>
    <td rowspan="2"> </td>
    <td colspan="2" align="center">LS</td>
   <td colspan="2" align="center">VCTK</td>
   <td rowspan="2">Avg.</td>
  </tr>
  <tr>
    <td>dev</td>
    <td>test</td>
    <td>dev</td>
    <td>test</td>
  </tr>
 <tr>
    <td>Char</td>
    <td>7.5</td>
    <td>7.4</td>
    <td>7.9</td>
    <td>11.7</td>
   <td>8.63</td>
  </tr>
 <tr>
    <td>BPE 100</td>
    <td>7.4</td>
    <td>7.2</td>
    <td><b>6.6</b></td>
    <td><b>10.7</b></td>
   <td><b>7.98</b></td>
  </tr>
 <tr>
    <td>BPE 200</td>
    <td><b>7.0</b></td>
    <td><b>6.9</b></td>
    <td>7.2</td>
    <td>11.1</td>
   <td>8.05</td>
  </tr>

</table>

File: egs2/libritts/asr1/local/phonemize_dir.py (hunk 2 above)
Comment (akreal) [reply egs2/libritts/asr1/local/phonemize_dir.py:22]: done!

**Comment by akreal** [comment]
Thanks for the review, @ftshijt !
I've addressed the comments and added README.

**Comment by ftshijt** [comment]
Looks very cool! Many thanks for your contribution.

**Pull request merged**